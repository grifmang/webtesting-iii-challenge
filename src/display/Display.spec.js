// Test away!
import React from "react";
import Display from "./Display";
import { render, getByText } from "@testing-library/react";

test('Check if Closed', () => {
    const { getByTestId, rerender } = render(<Display closed />);
    getByTestId('closedOpen');

    rerender(<Display closed={false} />);
    getByTestId('closedOpen');
})

test('Check if Locked', () => {
    const { getByTestId, rerender } = render(<Display locked />);
    getByTestId('lockedUnlocked');

    rerender(<Display locked={false} />);
    getByTestId('lockedUnlocked');
})

test('Test if green Green', () => {
    const className = 'led green-led';
    const locked = false;
    const closed = false;
    const display = render(<Display closed={closed} locked={locked} className={className} />)
    expect(display.baseElement).toMatchSnapshot();
})

test('Test if Red green', () => {
    const className = 'led red-led';
    const locked = false;
    const display = render(<Display closed locked={locked} className={className} />)
    expect(display.baseElement).toMatchSnapshot();
})

test('Test if red Red', () => {
    const className = 'led red-led';
    const locked = true;
    const closed = true;
    const display = render(<Display closed locked={locked} className={className} />)
    expect(display.baseElement).toMatchSnapshot();
})
