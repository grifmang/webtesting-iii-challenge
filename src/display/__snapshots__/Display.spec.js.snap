// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Red green 1`] = `
<body>
  <div>
    <div
      class="display panel"
    >
      <div
        class="led green-led"
        data-testid="lockedUnlocked"
      >
        Unlocked
      </div>
      <div
        class="led red-led"
        data-testid="closedOpen"
      >
        Closed
      </div>
    </div>
  </div>
</body>
`;

exports[`Test if green Green 1`] = `
<body>
  <div>
    <div
      class="display panel"
    >
      <div
        class="led green-led"
        data-testid="lockedUnlocked"
      >
        Unlocked
      </div>
      <div
        class="led green-led"
        data-testid="closedOpen"
      >
        Open
      </div>
    </div>
  </div>
</body>
`;

exports[`Test if red Red 1`] = `
<body>
  <div>
    <div
      class="display panel"
    >
      <div
        class="led red-led"
        data-testid="lockedUnlocked"
      >
        Locked
      </div>
      <div
        class="led red-led"
        data-testid="closedOpen"
      >
        Closed
      </div>
    </div>
  </div>
</body>
`;
